dependency/actions/compile/entrypoint

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env bash
+
+set -eu
+set -o pipefail
+
+DEST_DIR=$(mktemp -d)
+
+function main() {
+  local version output_dir target upstream_tarball working_dir
+  version=""
+  output_dir=""
+  target=""
+  upstream_tarball=""
+  working_dir=$(mktemp -d)
+
+  while [ "${#}" != 0 ]; do
+    case "${1}" in
+      --version)
+        version="${2}"
+        shift 2
+        ;;
+
+      --outputDir)
+        output_dir="${2}"
+        shift 2
+        ;;
+
+      --target)
+        target="${2}"
+        shift 2
+        ;;
+
+      "")
+        shift
+        ;;
+
+      *)
+        echo "unknown argument \"${1}\""
+        exit 1
+    esac
+  done
+
+  if [[ "${version}" == "" ]]; then
+    echo "--version is required"
+    exit 1
+  fi
+
+  if [[ "${output_dir}" == "" ]]; then
+    echo "--outputDir is required"
+    exit 1
+  fi
+
+  if [[ "${target}" == "" ]]; then
+    echo "--target is required"
+    exit 1
+  fi
+
+  # Only Ruby versions >= 3.1.0 work on Jammy
+  major=$(echo "${version%.*}" | cut -d '.' -f1)
+  minor=$(echo "${version%.*}" | cut -d '.' -f2)
+  if [[ "${target}" == "jammy" ]] && [[ ${major} -lt 3 || (${major} -eq 3 && ${minor} -lt 1) ]]; then
+    echo "version ${version} cannot be compiled on jammy, only 3.0.0 and above"
```

Comment:
We probably won't encounter this path, but we should ensure the error message is consistent with the logic.

The logic is >= 3.1.0 but the error message is >= 3.0.0

Reply:
this is a typo, I'll fix!